lib/posix/pthread_mutex.c

```diff
@@ -20,7 +20,7 @@ static const pthread_mutexattr_t def_attr = {
 	.type = PTHREAD_MUTEX_DEFAULT,
 };
 
-static int acquire_mutex(pthread_mutex_t *m, int timeout)
+static int acquire_mutex(pthread_mutex_t *m, k_timeout_t timeout)
```

Comment:
@carlescufi: So, the only way I can productively interpret comment "Use SYS_TIMEOUT_MS() instead which will convert SYS_FOREVER_MS to K_FOREVER automatically." below (outside pthread_rwlock.c, where it clearly makes sense) is as follows: "You can skip patching all the aux functions to accept k_timeout_t, instead can keep that at int/s32_t, and just use SYS_TIMEOUT_MS() inside such aux funcs to convert to k_timeout_t at the last mile."

If that's what you suggest, then:
1. How would I know that?
2. I specifically don't know that, as the whole idea seemed to be to use k_timeout_t as the primary timeout type. And I performed this refactor with that in mind. Then what you suggest is effectively along the lines of "you can ignore and workaround that k_timeout_t nuisance for as long as possible", so I'm not sure how good is it.

Reply:
No, I was only suggesting it in the places that needed a conversion from `s32_t` to `k_timeout_t` and where `-1` is a parameter, I wasn't suggesting that you stay with `s32_t`. Maybe I didn't express myself clearly enough, sorry about that.

Reply:
> No, I was only suggesting it in the places that needed a conversion from s32_t to k_timeout_t and where -1 is a parameter, I wasn't suggesting that you stay with s32_t. 

Good, then the current state of this PR should be exactly like that.